mig/images/js/jquery.filemanager.js

```diff
@@ -1013,6 +1013,53 @@ if (jQuery) (function($){
                   );
         }
 
+        function downloadWrapper(el, dialog, download_url) {
+            var lastinfo = $(statusinfo).html();
+
+            var target = $(el).attr(pathAttribute).split('/');
+            var file_name;
+            if (target.lastIndexOf('/') === target.length-1) {
+                file_name = 'output.bin';
+            } else {
+                file_name = target[target.length-1];
+            }
+            startProgress("Downloading "+file_name);
+
+            $.ajax({
+                url: download_url,
+                dataType: 'binary',
+                method: 'GET',
+                xhrFields: {
+                    responseType: 'blob' // Set the response type to 'blob'
+                },
+                success: function(blob) {
+                    console.info("Starting download of "+file_name)
+                    let fileURL = URL.createObjectURL(blob);
+                    var link = document.createElement('a');
+                    link.href = fileURL;
+                    /* downloaded filename */
+                    link.download = file_name;
+                    document.body.appendChild(link);
+                    link.click();
+                    URL.revokeObjectURL(fileURL);
+                    stopProgress("");
+                },
+                error: function(xhr, status, error) {
+                    console.error('Error downloading', file_name, ":" , error);
+                    //console.info('xhr: ', xhr);
+
+                    stopProgress("Download failed - above web size limit?");
```

Comment:
As far as I can tell this error handler will run for any failing request - but the only case for which this message is valid is IF the error came from the limit being enforced. In the absence of generic client side error handling infrastructure I'd expect to see some kind of conditional here and a fall back to a catch-all message about the download failing.

Reply:
Yes, it will run for _any_ error from the `cat.py` we point it to, so it overlaps somewhat with the TODO about digging out the actual error message from the binary output. That's the motivation for the vague progress message hint.